cartography/intel/gcp/compute.py

```diff
@@ -0,0 +1,118 @@
+            logger.debug(
+                (
+                    "Project %s returned a 404 not found error. "
+                    "Full details: %s"
+                ),
+                project_id,
+                e
+            )
+            return None
+        else:
+            logger.error("Could not use Compute Engine API on project %s; Reason: %s".format(project_id, reason))
+            raise e
+
+
+def get_gcp_instances_in_project(project_id, compute):
+    """
+    Return list of all GCP instances in a given project regardless of zone.
+    :param project_id: The project id
+    :param compute: The compute resource object created by googleapiclient.discovery.build()
+    :return: List of all GCP instances in given project regardless of zone.
+    """
+    zones = get_zones_in_project(project_id, compute)
+    if not zones:
+        # If the Compute Engine API is not enabled for a project, there are no zones and therefore no instances.
+        return []
+    instances = []
+    for zone in zones:
+        req = compute.instances().list(project=project_id, zone=zone['name'])
+        res = req.execute()
+        zone_instances = res.get('items', [])
+        for instance in zone_instances:
+            # Include project_id and zone_name on the `instance` dict so we can load it more easily
+            instance['project_id'] = project_id
+            instance['zone_name'] = zone['name']
+            instances.append(instance)
+    return instances
+
+
+def load_gcp_instances(neo4j_session, data, gcp_update_tag):
+    query = """
+    MATCH (p:GCPProject{id:{ProjectId}})
+    MERGE (i:GCPInstance{id:{InstanceId}})
+    ON CREATE SET i.firstseen = timestamp()
+    SET i.displayname = {DisplayName},
+    i.hostname = {Hostname},
+    i.zone_name = {ZoneName},
+    i.lastupdated = {gcp_update_tag}
+    WITH i, p
+    MERGE (p)-[r:RESOURCE]->(i)
+    ON CREATE SET r.firstseen = timestamp()
+    SET r.lastupdated = {gcp_update_tag}
+    """
+    for instance in data:
+        neo4j_session.run(
+            query,
+            ProjectId=instance['project_id'],
+            InstanceId=instance['id'],
+            DisplayName=instance.get('name', None),
+            Hostname=instance.get('hostname', None),
+            ZoneName=instance['zone_name'],
+            gcp_update_tag=gcp_update_tag
+        )
+
+
+def cleanup_gcp_instances(session, common_job_parameters):
+    run_cleanup_job('gcp_compute_instance_cleanup.json', session, common_job_parameters)
+
+
+def sync_gcp_instances(session, resources, gcp_update_tag, common_job_parameters):
+    crm_v1 = resources['crm_v1']
+    compute = resources['compute']
+
+    # Note: `crm.sync_gcp_projects()` calls `crm.get_gcp_projects()` before `compute.sync_gcp_instances()` does.
+    # We could run into an issue where the return values are different here.
+    # Still, we do need project_ids in order to pull instance data.
+    projects = get_gcp_projects(crm_v1)
+    for project in projects:
```

Comment:
How is this going to evolve when we add other GCP sync modules? I would suggest matching the AWS multi-account pattern unless there's a compelling reason otherwise. Ideally we call get_projects once and we avoid:
1. The issue you identified in the comment above.
2. The cross-dependency between compute and crm.